changelog/0077-sql-repeatrecord-pr1.yml

```diff
@@ -0,0 +1,103 @@
+title: 'SQL Repat Record Migration'
+key: sql-repeatrecord-pr1
+date: 2024-02-13
+optional_per_env: yes
+# (optional) Min version of HQ that MUST be deployed before this change can be rolled out (commit hash)
+min_commcare_version: TODO
+# (optional) Max version of HQ that can be deployed before this change MUST be rolled out (commit hash)
+max_commcare_version:
+context: |
+  Prepare for and migrate Repeat Records from Couch to SQL.
+
+details: |
+  This is an optional procedure that should be done if you use data forwarding
+  and want to ensure a future deploy is not blocked on needing to perform manual
+  migration steps.
+
+update_steps: |
+  Run the following command to determine how many repeat records you have to
+  migrate (in all commands below, replace `<env>` with your environment name):
+
+  ```sh
+  cchq <env> django-manage shell -c '
+  from corehq.motech.repeaters.management.commands.populate_repeatrecords import count_docs;
+  print("Records to migrate:", count_docs())'
+  ```
+
+  The output should include something like the following:
+
+  ```
+  Records to migrate: 125011
+  ```
+
+  A future deploy will be blocked, requiring manual migration steps, if the
+  number of records to migrate is more than 1000. These steps can be skipped
+  and the migration will be performed automatically during a future deploy if
+  the number is less than 1000 (as counted at deployment time). It is safe and
+  recommended to perform these steps even if the number is less than 1000.
+
+  Before proceeding, **make a full backup of your CommCare HQ data**. [Backup
+  procedures](https://commcare-cloud.readthedocs.io/en/latest/backups_dr/2-backups-guide.html)
+  are outside the scope of this document.
+
+  Next, estimate the amount of free space that will be required for new data in
+  PostgreSQL (main or repeaters database, depending on configuration):
+
+  ```sh
+  cchq <env> django-manage shell -c '
+  from corehq.motech.repeaters.models import RepeatRecord;
+  size = RepeatRecord.get_db().info()["sizes"]["active"];
+  mult = int((int.bit_length(size) - 1) / 10);
+  unit = " kMGT"[mult].strip();
```

Comment:
what is " kMGT" out of curiosity?

Reply:
Single-letter versions of kilo, Mega, Giga, Tera prefixes. I guess 'k' could have been capitalized.